lib/ds/id_generator.flow

```diff
@@ -0,0 +1,37 @@
+import maybe;
+import ds/tree;
+import string;
+
+export {
+	createStringIdGenerator(prefix : string) -> IdGenerator;
+	generateIntId() -> int;
+}
+
+intIdR = ref 0;
+
+generateIntId() -> int {
```

Comment:
Do I understand correctly, that intIdR is shared between all id generators and generateIntId, called from different usages, will generate unique ids? Sometimes it can be undesired behavior, since some ids at one specific usage would be missed.